scripts/_helpers.py

```diff
@@ -391,6 +391,38 @@ def _get_country(target, **keys):
         return np.nan
 
 
+def _getContinent(code):
+    """
+    Returns continent names that contains list of iso-code countries
+
+    Parameters
+    ----------
+    code : str
+        List of two letter country ISO codes
+
+    Returns
+    -------
+    continent_list : str
+        List of continent names
+
+    Example
+    -------
+    from helpers import _getContinent
+    code = ["DE", "GB", "NG", "ZA"]
+    _getContinent(code)
+    >>> ["africa", "europe"]
+    """
+    from config_osm_data import world_iso
```

Comment:
**Needs attention**

The repository for creating synthetic demand (https://github.com/euronion/synde/) does not follow the continent <-> country mapping and continent naming conventions from `config_osm_data`. As such this function will fail in some cases, e.g.:

* "Australia" in the demand data is part of the region `Oceania`, but the function would return `australia` as continent.
* "Mexico" is part of the region called "NorthAmerica", while the OSM data would return "north_america". Since this string is used to construct a file path, this does not match.

I recommend to use a dedicated function specifically for mapping between countries and regions (in the file names) from `synde`. The mapping can be accessed and downloaded from here https://github.com/euronion/synde/blob/master/config/regions.yaml or we can put it into the databundle the with demand data. @pz-max @davide-f can assist with the databundle?

Reply:
Looking at the data of [synde](https://github.com/euronion/synde/blob/master/config/regions.yaml), [our_world_iso](https://github.com/pypsa-meets-africa/pypsa-africa/blob/44b6236b641640b6502632df60773b30e4aa2f74/scripts/config_osm_data.py), and [wikipedia](https://en.wikipedia.org/wiki/List_of_sovereign_states_and_dependent_territories_by_continent_(data_file)), it seems our world_iso is not complete... We should update our world_iso to be complete and also adapt the correct continent naming

Reply:
Quick fix is to change our world_iso continent naming according to synde. Will be done in the next commit